src/SkyApm.Diagnostics.MassTransit/Observers/MasstransitConsumerObserver.cs

```diff
@@ -0,0 +1,85 @@
+﻿using MassTransit;
+using SkyApm.Common;
+using SkyApm.Config;
+using SkyApm.Diagnostics.MassTransit.Common;
+using SkyApm.Tracing;
+using SkyApm.Tracing.Segments;
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace SkyApm.Diagnostics.MassTransit.Observers
+{
+    public class MasstransitConsumerObserver : IConsumeObserver
+    {
+        private const string OperateNamePrefix = "Masstransit Consumer/";
+
+        private readonly ITracingContext _tracingContext;
+        private readonly IGetComponentUtil _getComponentID;
+        private TracingConfig _tracingConfig;
+        public MasstransitConsumerObserver(ITracingContext tracingContext,
+            IGetComponentUtil getComponentID,
+            IConfigAccessor configAccessor)
+        {
+            _tracingContext = tracingContext;
+            _getComponentID = getComponentID;
+            _tracingConfig = configAccessor.Get<TracingConfig>();
+        }
+        public Task PreConsume<T>(ConsumeContext<T> consumeContext) where T : class
+        {
+            var activity = Activity.Current ?? default;
+
+            var operationName = OperateNamePrefix + activity.OperationName;
+            var context = _tracingContext.CreateLocalSegmentContext(operationName);
```

Comment:
I think it would be more appropriate to use entryspan

Reply:
I have created the entryspan in ReceiveObserver that happened before the ConsumerObserver. If ConsumerObserver created the localspan which can GetParentSegmentContext created by ReceiveObserver.